Sprint 9 dz

AndrewkaW opened the pull request.

AndrewkaW committed:
Исприавил мелкие недочеты. в SettingsActivity перешел на ToolBar

AndrewkaW committed:
Добавил Темную тему и стили

AndrewkaW committed:
Реализовали 3 кнопик SettingsActivity

AndrewkaW committed:
Добвил поисковую строку на SearchActivity.kt, реализовал кнопку назад и кнопку очистки поисковой строки

AndrewkaW committed:
немного причесал activity_search.xml

faritowich commented on app/src/main/res/values/styles.xml:
```diff
@@ -0,0 +1,49 @@
+    <style name="MenuItemsStyle">
+        <item name="android:layout_width">match_parent</item>
+        <item name="android:layout_height">@dimen/height_of_points</item>
+        <item name="android:paddingHorizontal" tools:targetApi="o">@dimen/gen_indent</item>
```
Лучше отступы, высоту и ширину не выносить в стиль. Если на экране будет много элементов, будет сложнее ориентироваться с их взаимным расположением, когда отвечающие за это атрибуты разнесены по разным файлам - что-то может быть в лейауте, что-то в стиле. В стилях обычно содержат атрибуты, отвечающие за внешний вид вьюхи - цвета, шрифты, фоны и т.д.

faritowich commented on app/src/main/java/com/practicum/playlistmaker/SettingsActivity.kt:
```diff
@@ -1,16 +1,41 @@
+            toSupportIntent.data = Uri.parse("mailto:")
+            toSupportIntent.putExtra(Intent.EXTRA_EMAIL, arrayOf(getString(R.string.support_email)))
+            toSupportIntent.putExtra(Intent.EXTRA_TEXT,getString(R.string.support_massage))
+            startActivity(toSupportIntent)
```
Можно ещё использовать котлиновские функции области видимости, чтобы такие фрагменты кода сокращать до таких:

```

            Intent(Intent.ACTION_SENDTO).apply {
                data = Uri.parse("mailto:")
                putExtra(Intent.EXTRA_EMAIL, arrayOf(getString(R.string.support_email)))
                putExtra(Intent.EXTRA_TEXT,getString(R.string.support_massage))
                startActivity(this)
            }


```
Они иногда удобны, иногда нет, с практикой придёт.
Про такие методы почитать можно здесь:
https://kotlinlang.ru/docs/reference/scope-functions.html

faritowich commented on app/src/main/java/com/practicum/playlistmaker/SearchActivity.kt:
```diff
@@ -2,10 +2,86 @@ package com.practicum.playlistmaker
+        }
+    }
+
+    fun clearSearch() {
```
если метод будет использоваться только в рамках класса, в данном случае активити, то лучше делать его private

AndrewkaW committed:
Поправил верстку в соотвецтвии с макетом.
При нажатии на "Написать в поддержку" отправляется тема письма, как указано в задании. Цвет шрифта трех кнопок на экране настроек соответствует макету. Цвет Switch  в темнойц темен соответствует макету.  Цвет введенного текста на экране поиска соответствует макету. Применил все рекомендации к коду.

AndrewkaW committed:
поставил приватность у 2х функцуий в SearchActivity.kt
